packages/flutter_tools/lib/src/ios/core_devices.dart

```diff
@@ -79,14 +79,24 @@ class IOSCoreDeviceControl {
 
     try {
       final RunResult result = await _processUtils.run(command, throwOnError: true);
-
-      if (!output.existsSync()) {
+      final bool isPossiblyShutdown = _fileSystem is LocalFileSystem && _fileSystem.disposed;
+
+      // It's possible that the tool is in the process of shutting down, which
+      // could result in the temp directory being deleted after the shutdown hooks run
+      // before we check if `output` exists. If this happens, we shouldn't crash
+      // but just carry on as if no devices were found as the tool will exit on
+      // its own.
+      //
+      // See https://github.com/flutter/flutter/issues/141892 for details.
+      if (!isPossiblyShutdown && !output.existsSync()) {
```

Comment:
Nit: consider renaming `isPossiblyShutdown` to `isToolPossiblyShutdown`

Reply:
or `toolPossbilyShutdown`

Reply:
Done.